pandas/core/computation/eval.py

```diff
@@ -232,8 +237,23 @@ def eval(expr, parser='pandas', engine=None, truediv=True,
     pandas.DataFrame.query
     pandas.DataFrame.eval
     """
-    inplace = validate_bool_kwarg(inplace, 'inplace')
-    first_expr = True
+
+    def is_modifiable(obj):
+        target_type = type(obj).__name__
+        return hasattr(obj, "__dict__") or target_type == "dict"
+
+    inplace = validate_bool_kwarg(inplace, "inplace")
+
+    # "Primitives" cannot be modified inplace because
+    # they have no attributes that we can modify.
+    #
+    # The only exceptions are lists and dictionaries,
```

Comment:
@jorisvandenbossche : This is very much WIP.  I'm trying to test that the changes I make actually don't break anything, let alone > 200 tests like last time. :smile: